onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -34,23 +34,23 @@ bool ModelBuilder::IsNodeSupported(const Node& node) {
 }
 
 bool IsValidSupportedNodesVec(const std::vector<int>& supported_node_vec, const GraphViewer& graph_viewer) {
-  if (!supported_node_vec.empty()) {
-    if (supported_node_vec.size() == 1) {
-      const auto& node_indices = graph_viewer.GetNodesInTopologicalOrder();
-      const auto* node(graph_viewer.GetNode(node_indices[supported_node_vec[0]]));
-      const auto& op = node->OpType();
-      // It is not worth it to perform a single Reshape/Dropout/Identity operator
-      // which is only copying the data in NNAPI
-      // If this is the case, let it fall back
-      if (op == "Reshape" ||
-          op == "Dropout" ||
-          op == "Identity") {
-        return false;
-      }
+  if (!supported_node_vec.empty())
+    return false;
+
```

Comment:
If it's not empty we return false? If this is incorrect does a unit test somewhere fail?

Reply:
This is a typo for the last minute change, the UT won't fail since this mean the execution will fall back to CPU entirely
Is there any way we can verify the execution is on certain provider?

Reply:
After the model is loaded the nodes in the Graph should have been adjusted. Fused nodes are removed, and nodes have a value set in the execution provider field. We can get the graph from the inference session by using the InferenceSessionWrapper helper class in test utils.

---
In reply to: [508234410](https://github.com/microsoft/onnxruntime/pull/5545#discussion_r508234410) [](ancestors = 508234410)